airflow/providers/microsoft/azure/hooks/azure_data_factory.py

```diff
@@ -0,0 +1,674 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Any, Optional, Tuple
+
+from azure.mgmt.datafactory import DataFactoryManagementClient
+from azure.mgmt.datafactory.models import (
+    CreateRunResponse,
+    Dataset,
+    DatasetResource,
+    Factory,
+    LinkedService,
+    LinkedServiceResource,
+    PipelineResource,
+    PipelineRun,
+    Trigger,
+    TriggerResource,
+)
+from msrestazure.azure_operation import AzureOperationPoller
+
+from airflow.exceptions import AirflowException
+from airflow.providers.microsoft.azure.hooks.base_azure import AzureBaseHook
+
+
+class AzureDataFactoryHook(AzureBaseHook):
+    """
+    A hook to interact with Azure Data Factory.
+
+    :param conn_id: The Azure Data Factory connection id.
+    """
+
+    def __init__(self, conn_id: str = "azure_data_factory_default"):
+        super().__init__(sdk_client=DataFactoryManagementClient, conn_id=conn_id)
+
+        self._conn = self.get_conn()
+
+    def get_conn(self) -> DataFactoryManagementClient:
+        """
+        Return a Data Factory client.
+        """
+
+        conn = self.get_connection(self.conn_id)
+
+        self._resource_group_name = conn.extra_dejson.get("resourceGroup")
+        self._factory_name = conn.extra_dejson.get("factory")
+
+        return super().get_conn()
+
+    def _get_targeted_factory(
+        self, resource_group_name: Optional[str], factory_name: Optional[str]
+    ) -> Tuple[str, str]:
```

Comment:
You could also implement this as an annotation / decorator. 

Take a look at s3 for example:
- The creation: https://github.com/apache/airflow/blob/master/airflow/providers/amazon/aws/hooks/s3.py#L45-L65
- The call: https://github.com/apache/airflow/blob/master/airflow/providers/amazon/aws/hooks/s3.py#L140